### PR TITLE
Validate user ID in reclassification endpoint

### PR DESCRIPTION
**Pull request by lgonczar**
### Change description ###

Validate user ID in reclassification endpoint in order to prevent attacks through its value (user ID is logged). This should fix currently broken build (due to sonarcloud).

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/controllers/CcdCallbackController.java [hunk 1]
```diff
@@ -89,7 +89,9 @@ public CallbackResponse reclassifyExceptionRecord(
         @RequestBody CcdCallbackRequest callbackRequest,
         @RequestHeader(value = USER_ID, required = false) String userId
     ) {
-        if (callbackRequest.getCaseDetails() == null) {
+        if (userId != null && userId.matches("^.*[\\n|\\r|\\t]+.*$")) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/controllers/CcdCallbackController.java:92]: is this the only controller that should perform this validation?

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/controllers/CcdCallbackController.java:92]: Ideally, there should be more, but this one definitely needs that, as this user ID is logged. It's a quick fix. We may add this validation to other endpoints later.

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/controllers/CcdCallbackController.java:92]: do we use any validation api ?

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/controllers/CcdCallbackController.java:92]: we use vavr + custom validator for validating exception record details

**Pull request merged**